test/fixtures/test-runner/output/test-timeout-flag.js

```diff
@@ -0,0 +1,19 @@
+// Flags: --test-timeout=20
+'use strict';
+const { describe, test } = require('node:test');
+const { setTimeout } = require('node:timers/promises');
+
+describe('--test-timeout is set to 20ms', () => {
```

Comment:
I think this file is prone to being flaky since we are relying on relatively small timer differences in order for the test to work properly. This could be a problem in the CI.

Reply:
Since you want these tests to time out, I would make the `await setTimeout()` calls some really large value so they are guaranteed not to race in the CI. In other words, if the first two tests are supposed to time out after 20ms and 5ms, then have the `setTimeout()` be like 30 seconds. You could probably get away with not having a `setTimeout()` and have the tests never finish at all. For the third test, I would set the `timeout` option to something a lot higher too. For the final test, I would drop the `setTimeout()` completely.

Reply:
If I set it to a very large number e.g. 30 seconds, the test would fail after 20ms but because the event loop is still alive therefore the entire test execution would wait for 30 seconds, this might not be friendly to our CI I think?

Addressed for test 3 and test 4.

Reply:
ohh should've thought that! Thanks